packages/scripts/utils/index.js

```diff
@@ -1,99 +1,38 @@
-/**
- * External dependencies
- */
-const spawn = require( 'cross-spawn' );
-const { existsSync } = require( 'fs' );
-const path = require( 'path' );
-
 /**
  * Internal dependencies
  */
-const { getPackagePath, hasPackageProp } = require( './package' );
-const { exit, getCliArgs } = require( './process' );
-
-const getCliArg = ( arg ) => {
-	for ( const cliArg of getCliArgs() ) {
-		const [ name, value ] = cliArg.split( '=' );
-		if ( name === arg ) {
-			return value || null;
-		}
-	}
-};
-
-const hasCliArg = ( arg ) => getCliArg( arg ) !== undefined;
-
-const fromProjectRoot = ( fileName ) =>
-	path.join( path.dirname( getPackagePath() ), fileName );
-
-const hasProjectFile = ( fileName ) =>
-	existsSync( fromProjectRoot( fileName ) );
-
-const fromConfigRoot = ( fileName ) =>
-	path.join( path.dirname( __dirname ), 'config', fileName );
-
-const fromScriptsRoot = ( scriptName ) =>
-	path.join( path.dirname( __dirname ), 'scripts', `${ scriptName }.js` );
-
-const hasScriptFile = ( scriptName ) =>
-	existsSync( fromScriptsRoot( scriptName ) );
-
-const handleSignal = ( signal ) => {
-	if ( signal === 'SIGKILL' ) {
-		// eslint-disable-next-line no-console
-		console.log(
-			'The script failed because the process exited too early. ' +
-			'This probably means the system ran out of memory or someone called ' +
-			'`kill -9` on the process.'
-		);
-	} else if ( signal === 'SIGTERM' ) {
-		// eslint-disable-next-line no-console
-		console.log(
-			'The script failed because the process exited too early. ' +
-			'Someone might have called `kill` or `killall`, or the system could ' +
-			'be shutting down.'
-		);
-	}
-	exit( 1 );
-};
-
-const spawnScript = ( scriptName, args = [] ) => {
-	if ( ! scriptName ) {
-		// eslint-disable-next-line no-console
-		console.log( 'Script name is missing.' );
-		exit( 1 );
-	}
-
-	if ( ! hasScriptFile( scriptName ) ) {
-		// eslint-disable-next-line no-console
-		console.log(
-			'Unknown script "' + scriptName + '". ' +
-			'Perhaps you need to update @wordpress/scripts?'
-		);
-		exit( 1 );
-	}
-
-	const { signal, status } = spawn.sync(
-		'node',
-		[
-			fromScriptsRoot( scriptName ),
-			...args,
-		],
-		{ stdio: 'inherit' },
-	);
+const {
+	getCliArg,
+	getCliArgs,
+	hasCliArg,
+	spawnScript,
+} = require( './cli' );
+const {
+	fromConfigRoot,
+	hasProjectFile,
+} = require( './file' );
+const { hasPackageProp } = require( './package' );
 
-	if ( signal ) {
-		handleSignal( signal );
-	}
+const hasBabelConfig = () =>
+	hasProjectFile( '.babelrc' ) ||
+	hasProjectFile( 'babel.config.js' ) ||
+	hasPackageProp( 'babel' );
 
-	exit( status );
-};
+const hasJestConfig = () =>
+	hasCliArg( '-c' ) ||
+	hasCliArg( '--config' ) ||
+	hasProjectFile( 'jest.config.js' ) ||
+	hasProjectFile( 'jest.config.json' ) ||
+	hasPackageProp( 'jest' );
 
 module.exports = {
```

Comment:
This is missing `fromScriptsRoot`.

Reply:
Good catch, I think only `spawnScript` needs it at the moment so we might keep it internal. I will double check.

Reply:
It's ony used by `spawnScript` so we can leave it unexposed. Unless you think we should keep it exported?

Reply:
I don't feel strongly either way, I'd perhaps expose it for coherency, but I'm fine the other way round. Just wanted to bring our attention in case it was important.

Reply:
Can expose, it's fine 👍